### PR TITLE
BS-13323 input mappings on call activities

### PR DESCRIPTION
**Pull request by educhastenier**

### CODE REVIEW

File: bpm/bonita-api/bonita-common-api-impl/src/main/java/org/bonitasoft/engine/bpm/flownode/impl/internal/CallActivityDefinitionImpl.java [hunk 1]
```diff
@@ -98,6 +104,15 @@ public void setCallableElementType(final CallableElementType callableElementType
         this.callableElementType = callableElementType;
     }
 
+    public void addProcessStartContractInput(String inputName, Expression value) {
```
Comment (baptistemesta) [inline bpm/bonita-api/bonita-common-api-impl/src/main/java/org/bonitasoft/engine/bpm/flownode/impl/internal/CallActivityDefinitionImpl.java]: ProcessContractInput vs contractInput?

File: bpm/bonita-core/bonita-process-definition/bonita-process-definition-api/src/main/java/org/bonitasoft/engine/core/process/definition/model/SCallActivityDefinition.java [hunk 2]
```diff
@@ -34,4 +34,7 @@
 
     SCallableElementType getCallableElementType();
 
+    void addProcessStartContractInput(String key, SExpression value);
+
+    Map<String, SExpression> getProcessStartContractInputs();
```
Comment (baptistemesta) [inline bpm/bonita-core/bonita-process-definition/bonita-process-definition-api/src/main/java/org/bonitasoft/engine/core/process/definition/model/SCallActivityDefinition.java]: ProcessStartContractInput <-> ContractInput?

File: bpm/bonita-core/bonita-process-engine/src/main/java/org/bonitasoft/engine/execution/ProcessExecutorImpl.java [hunk 3]
```diff
@@ -163,15 +164,15 @@
     private final RefBusinessDataService refBusinessDataService;
 
     public ProcessExecutorImpl(final ActivityInstanceService activityInstanceService, final ProcessInstanceService processInstanceService,
-                               final TechnicalLoggerService logger, final FlowNodeExecutor flowNodeExecutor, final WorkService workService,
-                               final ProcessDefinitionService processDefinitionService, final GatewayInstanceService gatewayInstanceService,
-                               final TransitionService transitionService, final EventInstanceService eventInstanceService, final ConnectorService connectorService,
-                               final ConnectorInstanceService connectorInstanceService, final ClassLoaderService classLoaderService, final OperationService operationService,
+            final TechnicalLoggerService logger, final FlowNodeExecutor flowNodeExecutor, final WorkService workService,
```
Comment (baptistemesta) [inline bpm/bonita-core/bonita-process-engine/src/main/java/org/bonitasoft/engine/execution/ProcessExecutorImpl.java]: a lot of formartting, sometimes the PR add spaces here sometimes it removes some. Which one is correct?

File: bpm/bonita-api/bonita-common-api-impl/src/main/java/org/bonitasoft/engine/bpm/flownode/impl/internal/CallActivityDefinitionImpl.java (hunk 1 above)
Comment (educhastenier) [inline bpm/bonita-api/bonita-common-api-impl/src/main/java/org/bonitasoft/engine/bpm/flownode/impl/internal/CallActivityDefinitionImpl.java]: On call activity, we KNOW that it starts a process, so the naming can be precise. But later on, we may use the type ContractInput on something else than just Process start action.

File: bpm/bonita-core/bonita-process-definition/bonita-process-definition-api/src/main/java/org/bonitasoft/engine/core/process/definition/model/SCallActivityDefinition.java (hunk 2 above)
Comment (educhastenier) [inline bpm/bonita-core/bonita-process-definition/bonita-process-definition-api/src/main/java/org/bonitasoft/engine/core/process/definition/model/SCallActivityDefinition.java]: same answer

File: bpm/bonita-core/bonita-process-engine/src/main/java/org/bonitasoft/engine/execution/ProcessExecutorImpl.java (hunk 3 above)
Comment (educhastenier) [inline bpm/bonita-core/bonita-process-engine/src/main/java/org/bonitasoft/engine/execution/ProcessExecutorImpl.java]: The new version seems more correct to me, but may depend on your preferences...

**Pull request merged**